### DISCUSSION TITLE
Liquid Expressions not working with Dictionary<string, object> and also with UnsafeMemberAccessStrategy

### DISCUSSION THREAD

**Discussion by viraj-ldce** [elsa-workflows/elsa-core#4407] [Q&A]
Hi,

I am trying to use Liquid Expression for my custom class having Dictionary<string, object> as one property type. I have registered it with MemberAccessStrategy as well. This works if value in Dictionary is String but does not work if value in Dictionary is some complex type / class object.

**Registration**:
notification.TemplateContext.Options.MemberAccessStrategy.Register<MyCustomClass>();

**My Class**:
public class MyCustomClass
{
    public Dictionary<string, object> Record { get; set; }
}

I went through below documentation and tried to use **UnsafeMemberAccessStrategy** as mentioned but it did not work as well.
https://elsa-workflows.github.io/elsa-core/docs/next/expressions/expressions-liquid

Regards,
Viraj

**Comment by sfmskywalker** [comment]
Hello @viraj-ldce , I think that in those cases, you may have to register a [type interceptor](https://github.com/sebastienros/fluid#intercepting-a-type-access), but I'm not entirely sure. I recommend doing some experimentation and looking at other examples, found in the Elsa source code itself.
